src/index/vocabulary/PolymorphicVocabulary.h

```diff
@@ -87,6 +88,39 @@ class PolymorphicVocabulary {
         },
         vocab_);
   }
+
+  // Analogous to `lower_bound`, but since `word` is guaranteed to be a full
+  // word, not a prefix, this function can respect the split of an underlying
+  // `SplitVocabulary`.
+  template <bool getUpperBound, typename String, typename Comp>
+  WordAndIndex getPositionOfWord(const String& word, Comp comp) const {
+    return std::visit(
+        [&word, &comp](auto& vocab) {
+          using T = std::decay_t<decltype(vocab)>;
+          if constexpr (ad_utility::isInstantiation<T, SplitVocabulary>) {
```

Comment:
1. What I said above
2. in the else case make a `static_assert` that `T` is any of the remaining vocabularies withh a message that tells the user to think about the correct behavior when adding new vocab types to the Vocab.